src/v/redpanda/application.cc

```diff
@@ -613,6 +613,18 @@ void application::start() {
     vlog(
       _log.info, "Started Kafka API server listening at {}", conf.kafka_api());
 
+    /// Start client listening for events on the internal coprocessor topic
+    if (coproc_enabled()) {
+        /// Temporarily disable retries for the new client until we create a
+        /// more granular way to configure this per client or per request.
+        kafka::client::shard_local_cfg().retries.set_value(size_t(0));
```

Comment:
It would be good if we move away from shard local configurations and instead use objects. it makes things very hard to test.

Reply:
Yeah @dotnwat is making the change currently

Reply:
this is where i got to https://github.com/dotnwat/redpanda/commit/deb7b5bec81e3bc2e434bc054685c11c5d929be8 maybe @BenPope has an opinion on it

Reply:
This is my suggestion for now. It's not pretty, but it's easy to find for when the config improvements land.

Some alternatives:
* Add a retry parameter to requests.
* Fix the config to be passed by value to the client, and provide an API for changing it, so that changing the value during testing is possible and maybe expose that via the http admin interface.

I like the first one, and will likely do it, but it's not that high on my priority list, unless it's a blocker for somebody (and PRs welcome, etc.).

The second one isn't likely to happen before this lands.

If coproc is ok handling all the retry logic externally, for all requests, then this change is the right balance.

Reply:
@BenPope yeah its been hydrated as this happens in `wire_up_services()`. I believe this is fine and desired for coproc now as it stands because its only using the fetch request and the call exists in a poll forever loop, so if there are any errors , retry logic will implicitly occur anyway.